salt/transport/client.py

```diff
@@ -38,15 +42,25 @@ def crypted_transfer_decode_dictentry(self, load, dictkey=None, tries=3, timeout
         '''
         raise NotImplementedError()
 
+    def __enter__(self):
+        return self
+
+    def __exit__(self):
```

Comment:
Beware of this:
https://docs.quantifiedcode.com/python-anti-patterns/correctness/exit_must_accept_three_arguments.html

It won't get called!

Reply:
Yep, I rushed through that and forgot about the `__exit__` signature.